.github/workflows/check.yaml

```diff
@@ -26,18 +26,49 @@ jobs:
       tox-version: "<4"
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
     needs: lint-unit
+    name: functional tests
+    runs-on: ${{ matrix.architecture.runs-on }}
+    timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
-        include:
-          - juju-channel: "3.4/stable"
-            command: "TEST_JUJU3=1 make functional"  # using TEST_JUJU3 due https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6
-    with:
-      command: ${{ matrix.command }}
-      juju-channel: ${{ matrix.juju-channel }}
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+        juju-channel: ["3.4/stable"]
+        architecture:
+          - runs-on: [ubuntu-latest]  # github hosted runners are amd64
+            model-constraints: ""
+            # TODO: should we use these or the Ubuntu_ARM64_4C_16G_01 github-hosted runner?
```

Comment:
This is pending internal discussion.

Reply:
Should we use also size? Not sure if for ARM64 we needed such label, but for amd64 we are using it.

Reply:
This is the label of github official arm64 runner, we will need it if our decision is to use github official one.

Reply:
If there are no objections, I'm going to leave it as-is for now.  We can easily change it later and be consistent across projects once it's managed via terraform.  It's working for now. :)